### ISSUE TITLE
Add option to disable tag parsing

### ISSUE THREAD

**Issue by kien-truong** [prometheus/statsd_exporter#324]
Since tagging is not standardized in statsd, the parser can get confused if there are characters like `,` or `#` in metrics name when attempting to detect tagging styles. 

We should support processing the metric's name as-is and not parsing tags altogether.

**Comment by matthiasr** [comment]
yes, this makes sense. Should this be one flag to disable them all, or separate flags for each tag format?

**Comment by kien-truong** [comment]
I think the easiest and most backward-compatible way would be adding a flag to disable all tag parsing. Users would not have to change anything if they want to keep the current behavior.

However, since mixing multiple tag styles is not supported, may be we should ask the user to explicitly choose the one tagging style they want to use. This option also makes it easier to support more tagging styles in the future.

**Comment by matthiasr** [comment]
I don't think a global on/off would be backward compatible – we recently added a new tagging style, and with a single flag you would have to turn off the older styles (that you maybe use) while keeping the others.

It is also valid (and I expect that it happens) that the same exporter receives metrics in different styles – we only do not support mixing them within one submission, so I would not want to make users choose only one.

I'm thinking we could add a boolean flag for each tagging style, and default them all to "on". I think enabling everything by default is the most user friendly (and by now, backwards compatible) behavior here but happy to hear more opinions on that.
